File: JavaScript/1823_Find_the_Winner_of_the_Circular_Game/solution.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number}
 */
var findTheWinner = function (n, k) {
    const friends = new Array(n);
    for (let i = 0; i < n; i++) {
        friends[i] = i + 1;
    }

    let complement = 0;
    while (friends.length > 1) {
        const lose = (complement + k) % friends.length;
        friends.splice(lose - 1, 1);
        complement = lose - 1;
        if (complement < 0) {
            complement = friends.length;
        }
    }

    return friends[0];
};

File: Go/1823_Find_the_Winner_of_the_Circular_Game/solution.go
package solution

func findTheWinner(n int, k int) int {
	var friends []int
	for i := 1; i <= n; i++ {
		friends = append(friends, i)
	}

	for len(friends) > 1 {
		for i := 0; i < k-1; i++ {
			temp := friends[0]
			friends = append(friends[1:], temp)
		}

		friends = friends[1:]
	}

	return friends[0]
}
